tests/integration-tests-isolated/src/test/java/org/apache/activemq/artemis/tests/integration/isolated/client/ConnectionDroppedTest.java

```diff
@@ -0,0 +1,508 @@
+      ReusableLatch latchCreating = new ReusableLatch(1);
+      ReusableLatch blockCreate = new ReusableLatch(1);
+      ReusableLatch done = new ReusableLatch(1);
+      ActiveMQServer server = createServer(true, createDefaultConfig(true));
+      server.start();
+
+      int TEST_REPEATS = 4;
+
+      server.registerBrokerPlugin(new ActiveMQServerSessionPlugin() {
+         @Override
+         public void beforeCreateSession(String name,
+                                         String username,
+                                         int minLargeMessageSize,
+                                         RemotingConnection connection,
+                                         boolean autoCommitSends,
+                                         boolean autoCommitAcks,
+                                         boolean preAcknowledge,
+                                         boolean xa,
+                                         String defaultAddress,
+                                         SessionCallback callback,
+                                         boolean autoCreateQueues,
+                                         OperationContext context,
+                                         Map<SimpleString, RoutingType> prefixes) throws ActiveMQException {
+            latchCreating.countDown();
+            try {
+               blockCreate.await(10, TimeUnit.HOURS);
+            } catch (Exception e) {
+               logger.warn(e.getMessage(), e);
+            } finally {
+               done.countDown();
+            }
+         }
+      });
+
+      AtomicBoolean running = new AtomicBoolean(true);
+      ExecutorService executorService = Executors.newFixedThreadPool(1);
+      runAfter(executorService::shutdownNow);
+      runAfter(() -> running.set(false));
+      Queue serverQueue = server.createQueue(new QueueConfiguration(getName()).setRoutingType(RoutingType.ANYCAST).setAddress(getName()).setAutoCreated(false));
+
+      for (int i = 0; i < TEST_REPEATS; i++) {
+         Assert.assertEquals(0, serverQueue.getConsumerCount());
+         latchCreating.setCount(1);
+         blockCreate.setCount(1);
+         done.setCount(1);
+
+         ProtonTestClient peer = new ProtonTestClient();
+
+         executorService.execute(() -> {
+
+            try {
+               peer.queueClientSaslAnonymousConnect();
+               peer.remoteOpen().queue();
+               peer.remoteBegin().queue();
+               peer.remoteAttach().ofReceiver().withName(RandomUtil.randomString()).withSenderSettleModeUnsettled().withReceivervSettlesFirst().withTarget().also().withSource().withAddress(getName()).withExpiryPolicyOnLinkDetach().withDurabilityOfNone().withCapabilities("queue").withOutcomes("amqp:accepted:list", "amqp:rejected:list").also().queue();
+
+               peer.connect("localhost", 61616);
+
+               // Waits for all the commands to fire and the drop action to be run.
```

Comment:
Comment is not correct, seems to be cut and pasted from another test, no drop action was added in this test.